src/packages/core/property-editor/uis/tree-picker/config/filter/property-editor-ui-tree-picker-filter.element.ts

```diff
@@ -0,0 +1,95 @@
+import { UMB_DATA_TYPE_WORKSPACE_CONTEXT } from '../../../../../data-type/workspace/data-type-workspace.context.js';
+import { UmbDocumentTypeInputElement } from '../../../../../../documents/document-types/components/document-type-input/document-type-input.element.js';
+import { UmbMediaTypeInputElement } from '../../../../../../media/media-types/components/media-type-input/media-type-input.element.js';
+import { StartNode } from '@umbraco-cms/backoffice/components';
+import type { UmbPropertyEditorUiElement } from '@umbraco-cms/backoffice/extension-registry';
+import { customElement, html, property } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+
+@customElement('umb-property-editor-ui-tree-picker-filter')
+export class UmbPropertyEditorUITreePickerFilterElement extends UmbLitElement implements UmbPropertyEditorUiElement {
+	constructor() {
+		super();
+
+		this.consumeContext(UMB_DATA_TYPE_WORKSPACE_CONTEXT, (instance) => {
+			const workspace = instance;
+			this.observe(workspace.data, (data) => {
+				const property = data?.values.find((setting) => setting.alias === 'startNode');
+				if (property) {
+					const startNode = property.value as StartNode;
+					if (startNode.type) {
+						this.sourceType = startNode.type;
+					}
+				}
+			});
+		});
+	}
+
+	@property({ type: Array })
+	value?: string[];
+
+	@property({ attribute: false })
+	sourceType: string = 'content';
```

Comment:
`sourceType` is a internal property. So you should mark it with `private` then TypeScript will complain if someone tries to access this. (we also have JS private fields, that is marked with # as the first character, but these does not cope with Lit. So we just use TypeScript for these.)

Also since this is internal, we can spare some performance and just make the property a Lit `@state()` thats equivalent to `@property` just internal. (where property comes with features to be changed from the outside, sync with attribute and all that)

Reply:
👍 I've updated the property.